src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -67,6 +68,13 @@ def __init__(self,  # pylint: disable=too-many-arguments
         self.active_directory_resource_id = active_directory_resource_id
         self.active_directory_graph_resource_id = active_directory_graph_resource_id
 
+    def has_endpoint_set(self, endpoint_name):
+        try:
+            getattr(self, endpoint_name)
```

Comment:
The behavior for `hasattr` for [Python 2](https://docs.python.org/2/library/functions.html#hasattr) vs [Python 3](https://docs.python.org/3/library/functions.html#hasattr) is *slightly* different.
We actually want the Python 2 implementation.
I'll add a comment in the code to clarify.